src/vs/workbench/contrib/codeEditor/browser/saveParticipants.ts

```diff
@@ -277,15 +277,11 @@ class CodeActionOnSaveParticipant implements ITextFileSaveParticipant {
 			return;
 		}
 
-		// Do not run code actions on auto save
-		if (env.reason !== SaveReason.EXPLICIT) {
-			return undefined;
-		}
-
 		const textEditorModel = model.textEditorModel;
+		const settingsOverrides = { overrideIdentifier: textEditorModel.getLanguageId(), resource: textEditorModel.uri };
 
-		const settingsOverrides = { overrideIdentifier: textEditorModel.getLanguageId(), resource: model.resource };
-		const setting = this.configurationService.getValue<{ [kind: string]: boolean } | string[]>('editor.codeActionsOnSave', settingsOverrides);
+		// Keeping string | boolean until fully deprecating boolean options
+		const setting = this.configurationService.getValue<{ [kind: string]: string | boolean } | string[]>('editor.codeActionsOnSave', settingsOverrides);
 		if (!setting) {
 			return undefined;
 		}
```

Comment:
Instead of handling the boolean values below, can you add logic here that converts the old boolean settings to the new enum values: `false -> "never"`, `true -> "explict"`. That should simplify the logic